sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/Rules/CreateRuleOptions.cs

```diff
@@ -119,15 +119,10 @@ public override bool Equals(object obj)
         /// <inheritdoc/>
         public bool Equals(CreateRuleOptions other)
         {
-            if (other is CreateRuleOptions otherOptions
-                && string.Equals(Name, otherOptions.Name, StringComparison.OrdinalIgnoreCase)
-                && (Filter == null || Filter.Equals(otherOptions.Filter))
-                && (Action == null || Action.Equals(otherOptions.Action)))
-            {
-                return true;
-            }
-
-            return false;
+            return other is not null
+                   && string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase)
+                   && (Filter == null || Filter.Equals(other.Filter))
```

Comment:
Technically this could be written as `Filter?.Equals(other.Filter) != false`. Thoughts?

Reply:
I'm good either way.   If I was writing it fresh, I'd use the form in your suggestion.

Reply:
Ok pushed. I'd write it that way too.